.github/CONTRIBUTING.md

```diff
@@ -28,10 +64,40 @@ pnpm run format
 
 ## Running tests
 
-You can run the test suite using the following commands:
+You can run the TypeScript and Rust test suites using the following command:
+
+```sh
+pnpm test
+```
+
+To run the integration tests, use:
+
+```sh
+pnpm build && pnpm test:integrations
+```
+
+Additionally, some features require testing in browsers (i.e to ensure CSS variable resolution works as expected). These can be run via:
 
 ```sh
-pnpm build && pnpm test
+pnpm build && pnpm test:ui
 ```
 
-Please ensure that the tests are passing when submitting a pull request. If you're adding new features to Tailwind, please include tests.
+Please ensure that all tests are passing when submitting a pull request. If you're adding new features to Tailwind CSS, always include tests.
```

Comment:
I think we should mention again that new features require some discussion first. Something similar to what we have in the PR template:

> It's never a fun experience to have your pull request declined after investing a lot of time and effort into a new feature. To avoid this from happening, we request that contributors create ~~an issue~~ **a discussion** to first discuss any significant new features.

The only problem with the PR template is that you typically only see that when opening a PR... aka, when you already worked on your feature.